msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerAuthServiceStrategy.java

```diff
@@ -253,50 +240,26 @@ static Bundle getRequestBundleForGetAccounts(@NonNull final OperationParameters
     }
 
     @WorkerThread
-    protected boolean removeBrokerAccount(@NonNull final OperationParameters parameters) throws BaseException, InterruptedException, ExecutionException, RemoteException {
+    protected void removeBrokerAccount(@NonNull final OperationParameters parameters)
+            throws BaseException, InterruptedException, ExecutionException, RemoteException {
         final String methodName = ":removeBrokerAccountWithAuthService";
-        Telemetry.emit(
-                new BrokerStartEvent()
-                        .putAction(methodName)
-                        .putStrategy(TelemetryEventStrings.Value.BOUND_SERVICE)
-        );
-
-        IMicrosoftAuthService service;
-        final MicrosoftAuthClient client = new MicrosoftAuthClient(parameters.getAppContext());
-
-        try {
-            final MicrosoftAuthServiceFuture authServiceFuture = client.connect();
-
-            service = authServiceFuture.get();
-
-            Bundle requestBundle = getRequestBundleForRemoveAccount(parameters);
-            service.removeAccount(requestBundle);
-            Telemetry.emit(
-                    new BrokerEndEvent()
-                            .putAction(methodName)
-                            .isSuccessful(true)
-            );
-
-            return true;
-        } catch (final BaseException | InterruptedException | ExecutionException | RemoteException e) {
-            com.microsoft.identity.common.internal.logging.Logger.error(
-                    TAG + methodName,
-                    "Exception is thrown when trying to get target account."
-                            + e.getMessage(),
-                    ErrorStrings.IO_ERROR,
-                    e);
-            Telemetry.emit(
-                    new BrokerEndEvent()
-                            .putAction(methodName)
-                            .isSuccessful(false)
-                            .putErrorCode(ErrorStrings.IO_ERROR)
-                            .putErrorDescription(e.getLocalizedMessage())
-            );
-
-            throw e;
-        } finally {
-            client.disconnect();
-        }
+        performAuthServiceOperation(parameters.getAppContext(),
+                new AuthServiceOperation<Void>() {
+                    @Override
+                    public Void perform(IMicrosoftAuthService service) throws RemoteException, BaseException {
+                        final Bundle requestBundle = getRequestBundleForRemoveAccount(parameters);
+                        MsalBrokerResultAdapter.verifyRemoveAccountResultFromBundle(
+                                service.removeAccount(requestBundle)
+                        );
+
+                        return null;
+                    }
+
+                    @Override
+                    public String getOperationName() {
+                        return methodName;
```

Comment:
Why use methodName here whereas hard coding in all the above methods. I think we can be consistent here.